Disclaimer on state modified behavior change (#6357)
fishtown-analytics/docs.getdbt.com: website/snippets/_state-modified-compare.md

> The state directory needs to be built with dbt v1.9 or higher, or [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud along with `state_modified_compare_more_unrendered_values: true` within your dbt_project.yml.

Reviewer: Changing to make this easier to read.

```suggestion
The state directory needs to be built with dbt v1.9 or higher, or [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud along with setting `state_modified_compare_more_unrendered_values` to true within your dbt_project.yml.
```

> The state directory needs to be built with dbt v1.9 or higher, or [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud along with setting `state_modified_compare_more_unrendered_values` to true within your dbt_project.yml.

Reviewer: ```suggestion
You need to build the state directory using dbt v1.9 or higher, or [Versionless](/docs/dbt-versions/versionless-cloud) dbt Cloud, and you need to set `state_modified_compare_more_unrendered_values` to `true` within your dbt_project.yml.
```

> If the state directory was built with an older dbt version or if the `state_modified_compare_more_unrendered_values` behavior change flag was not set (or it was set to `false`), you need to rebuild the state directory to avoid false positives during state comparison with `state:modified`.

Reviewer: ```suggestion
If the state directory is built with an older dbt version or if the `state_modified_compare_more_unrendered_values` behavior change flag was either not set or set to `false`, you need to rebuild the state directory to avoid false positives during state comparison with `state:modified`.
```

> If the state directory is built with an older dbt version or if the `state_modified_compare_more_unrendered_values` behavior change flag was either not set or set to `false`, you need to rebuild the state directory to avoid false positives during state comparison with `state:modified`.

Reviewer: ```suggestion
If the state directory was built with an older dbt version or if the `state_modified_compare_more_unrendered_values` behavior change flag was either not set or set to `false`, you need to rebuild the state directory to avoid false positives during state comparison with `state:modified`.
```
